src/Fantomas.Core/CodePrinter.fs

```diff
@@ -1695,6 +1557,157 @@ let genQuoteExpr (node: ExprQuoteNode) =
+
+                    // Add spaces to ensure the record field (incl trivia) starts at the right column.
+                    addFixedSpaces targetColumn
+                    +> atCurrentColumn (enterNode fieldNode +> genIdentListNode fieldNode.FieldName)
+                    +> sepSpace
+                    +> genSingleTextNode fieldNode.Equals
+                    +> genNode (genExpr fieldNode.Expr)
+                    +> leaveNode fieldNode)
+                ctx
+
+let genMultilineStandardRecordExprCrampedFields (node: ExprCopyableRecordNode) targetColumn =
+    match node.CopyInfo with
+    | Some we -> genCopyExpr (genFieldsExpr node) we
+    | None ->
+        fun (ctx: Context) ->
+            col
+                sepNln
+                node.Fields
+                (fun e ->
+                    // Add spaces to ensure the record field (incl trivia) starts at the right column.
+                    addFixedSpaces targetColumn
+                    // Lock the start of the record field, however keep potential indentations in relation to the opening curly brace
+                    +> atCurrentColumn (genRecordFieldName e))
+                ctx
+
+let genMultilineRecord genCrampedFields (node: ExprCopyableRecordNode) (ctx: Context) =
+    let expressionStartColumn = ctx.Column
+
+    let targetColumn =
+        let openBracketLength = node.OpeningBrace.Text.Length
+
+        expressionStartColumn
+        + (if ctx.Config.SpaceAroundDelimiter then
+               openBracketLength + 1
+           else
+               openBracketLength)
+
+    let genMultilineCramped =
+        match node.CopyInfo with
+        | Some _ ->
+            genSingleTextNode node.OpeningBrace
+            +> sepNlnWhenWriteBeforeNewlineNotEmptyOr addSpaceIfSpaceAroundDelimiter // comment after curly brace
+            +> genCrampedFields targetColumn
+            +> addSpaceIfSpaceAroundDelimiter
+            +> genSingleTextNode node.ClosingBrace
+        | None ->
+            atCurrentColumn (
+                genSingleTextNodeSuffixDelimiter node.OpeningBrace
+                +> sepNlnWhenWriteBeforeNewlineNotEmpty // comment after curly brace
+                +> genCrampedFields targetColumn
+                +> sepNlnWhenWriteBeforeNewlineNotEmpty
+                +> (fun ctx ->
+                    // Edge case scenario to make sure that the closing brace is not before the opening one
+                    // See unit test "multiline string before closing brace"
+                    let brace =
+                        addFixedSpaces expressionStartColumn +> genSingleTextNode node.ClosingBrace
+
+                    ifElseCtx lastWriteEventIsNewline brace (addSpaceIfSpaceAroundDelimiter +> brace) ctx)
+            )
+
+    ifAlignOrStroustrupBrackets (genMultilineRecordAlignBrackets node) genMultilineCramped ctx
+
+let genRecord smallRecordExpr multilineRecordExpr (node: ExprRecordNode) =
```

Comment:
Add `(ctx: Context)` as a last parameter for `genRecord` instead of returning a lambda.